internal/js/modules/k6/browser/tests/page_test.go

```diff
@@ -2787,3 +2786,102 @@ func TestPageMustUseNativeJavaScriptObjects(t *testing.T) {
 	_, err = page.QueryAll("#textField")
 	require.NoErrorf(t, err, "page should not override the native objects, but it did")
 }
+
+func TestWaitForNavigationWithURL(t *testing.T) {
+	t.Parallel()
+
+	tb := newTestBrowser(t, withFileServer())
+	tb.vu.ActivateVU()
+	tb.vu.StartIteration(t)
+
+	got := tb.vu.RunPromise(t, `
+		const page = await browser.newPage();
+		const testURL = '%s';
+
+		try {
+			await page.goto(testURL);
+
+			// Test exact URL match
+			await Promise.all([
+				page.waitForNavigation({ url: '%s' }),
+				page.click('#page1')
+			]);
+			let currentURL = page.url();
+			if (!currentURL.endsWith('page1.html')) {
+				throw new Error('Expected to navigate to page1.html but got ' + currentURL);
+			}
+
+			await page.goto(testURL);
+
+			// Test regex pattern - matches any page with .html extension
+			await Promise.all([
+				page.waitForNavigation({ url: /.*\.html$/ }),
+				page.click('#page2')
+			]);
+			currentURL = page.url();
+			if (!currentURL.endsWith('.html')) {
+				throw new Error('Expected URL to end with .html but got ' + currentURL);
+			}
+
+			await page.goto(testURL);
+
+			// Test timeout when URL doesn't match
+			let timedOut = false;
+			try {
+				await Promise.all([
+					page.waitForNavigation({ url: /.*nonexistent.html$/, timeout: 500 }),
+					page.click('#page1')  // This goes to page1.html, not nonexistent.html
+				]);
+			} catch (error) {
+				if (error.toString().includes('waiting for navigation')) {
+					timedOut = true;
+				} else {
+					throw error;
+				}
+			}
+			if (!timedOut) {
+				throw new Error('Expected timeout error when URL does not match');
+			}
+
+			await page.goto(testURL);
+
+			// Test empty pattern (matches any navigation)
+			await Promise.all([
+				page.waitForNavigation({ url: '' }),
+				page.click('#page2')
+			]);
+			currentURL = page.url();
+			if (!currentURL.endsWith('page2.html')) {
+				throw new Error('Expected empty pattern to match any navigation but got ' + currentURL);
+			}
+		} finally {
+			// Must call close() which will clean up the taskqueue.
+			await page.close();
+		}
+	`,
+		tb.staticURL("waitfornavigation_test.html"),
+		tb.staticURL("page1.html"),
+	)
+	assert.Equal(t, sobek.Undefined(), got.Result())
+}
+
+func TestWaitForNavigationWithURL_RegexFailure(t *testing.T) {
```

Comment:
@joanlopez I created this test, but it's really only testing the Sobek side of things. Invalid regex don't manage to get passed to the `waitForURL` API itself.

Reply:
So, I guess the error returned by Sobek (when calling `RunAsync`), would be possible to `catch` and `console.log`, right? 🤔  Thanks!